plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/move/MoveTypeChange.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Mario Kastner
+ *     - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+
+package org.eclipse.fordiac.ide.typemanagement.refactoring.move;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.fordiac.ide.model.IdentifierVerifier;
+import org.eclipse.fordiac.ide.model.commands.change.ChangePackageNameCommand;
+import org.eclipse.fordiac.ide.model.helpers.PackageNameHelper;
+import org.eclipse.fordiac.ide.model.libraryElement.LibraryElement;
+import org.eclipse.fordiac.ide.model.typelibrary.TypeEntry;
+import org.eclipse.fordiac.ide.typemanagement.refactoring.AbstractCommandChange;
+import org.eclipse.fordiac.ide.typemanagement.refactoring.IFordiacModifiablePreviewChange;
+import org.eclipse.fordiac.ide.typemanagement.wizards.Messages;
+import org.eclipse.gef.commands.Command;
+import org.eclipse.ltk.core.refactoring.RefactoringStatus;
+
+public class MoveTypeChange extends AbstractCommandChange<LibraryElement> implements IFordiacModifiablePreviewChange {
+
+	private String newPackageName;
+	private String oldPackageName;
+	private final TypeEntry type;
+
+	protected MoveTypeChange(final String newPackageName, final TypeEntry type, final String name, final URI uri,
+			final URI destination) {
+		super(name, uri, LibraryElement.class);
+		this.newPackageName = newPackageName;
+		this.type = type;
+	}
+
+	@Override
+	public void initializeValidationData(final LibraryElement element, final IProgressMonitor pm) {
+		oldPackageName = PackageNameHelper.getPackageName(element);
+	}
+
+	@Override
+	public RefactoringStatus isValid(final LibraryElement element, final IProgressMonitor pm)
+			throws CoreException, OperationCanceledException {
+		final RefactoringStatus status = new RefactoringStatus();
+		if (!Objects.equals(element.getTypeEntry().getPackageName(), oldPackageName)) {
```

Comment:
Please check the element instead of the cached value from the type entry:
```suggestion
		if (!Objects.equals(PackageNameHelper.getPackageName(element), oldPackageName)) {
```